build/lib/extensions.js

```diff
@@ -235,68 +235,47 @@ function packageMarketplaceExtensionsStream() {
-    const visxDirectory = path.join(path.dirname(root), 'vsix');
-    try {
-        if (!fs.existsSync(visxDirectory)) {
-            fs.mkdirSync(visxDirectory);
-        }
-    }
-    catch (err) {
-        // don't fail the build if the output directory already exists
-        console.warn(err);
-    }
-    sqlBuiltInLocalExtensionDescriptions.forEach(element => {
-        let pkgJson = JSON.parse(fs.readFileSync(path.join(element.path, 'package.json'), { encoding: 'utf8' }));
-        const packagePath = path.join(visxDirectory, `${pkgJson.name}-${pkgJson.version}.vsix`);
-        console.info('Creating vsix for ' + element.path + ' result:' + packagePath);
-        vsce.createVSIX({
-            cwd: element.path,
-            packagePath: packagePath,
-            useYarn: true
-        });
-    });
-}
-exports.packageSQLExtensions = packageSQLExtensions;
-function packageExtensionTask(extensionName, platform, arch) {
-    var destination = path.join(path.dirname(root), 'azuredatastudio') + (platform ? '-' + platform : '') + (arch ? '-' + arch : '');
-    if (platform === 'darwin') {
-        destination = path.join(destination, 'Azure Data Studio.app', 'Contents', 'Resources', 'app', 'extensions', extensionName);
-    }
-    else {
-        destination = path.join(destination, 'resources', 'app', 'extensions', extensionName);
-    }
-    platform = platform || process.platform;
-    return () => {
-        const root = path.resolve(path.join(__dirname, '../..'));
-        const localExtensionDescriptions = glob.sync('extensions/*/package.json')
+    return new Promise((resolve, reject) => {
+        const sqlBuiltInLocalExtensionDescriptions = glob.sync('extensions/*/package.json')
             .map(manifestPath => {
             const extensionPath = path.dirname(path.join(root, manifestPath));
             const extensionName = path.basename(extensionPath);
             return { name: extensionName, path: extensionPath };
         })
-            .filter(({ name }) => extensionName === name);
-        const localExtensions = es.merge(...localExtensionDescriptions.map(extension => {
-            return fromLocal(extension.path);
-        }));
-        let result = localExtensions
-            .pipe(util2.skipDirectories())
-            .pipe(util2.fixWin32DirectoryPermissions())
-            .pipe(filter(['**', '!LICENSE', '!LICENSES.chromium.html', '!version']));
-        return result.pipe(vfs.dest(destination));
-    };
+            .filter(({ name }) => excludedExtensions.indexOf(name) === -1)
+            .filter(({ name }) => builtInExtensions.every(b => b.name !== name))
+            .filter(({ name }) => sqlExtensions.indexOf(name) >= 0);
+        es.merge(sqlBuiltInLocalExtensionDescriptions.map(extension => {
+            return fromLocal(extension.path)
+                .pipe(rename(p => p.dirname = `extensions/${extension.name}/${p.dirname}`));
+        }).map(p => p.pipe(gulp.dest('.build/external')))).on('end', () => {
```

Comment:
A bunch of optimization here to reuse as much code as we can from vscode's extension compile steps. probably some more optimizations could be made here to reduce this code.

Benefit of this change is it now can utilize the webpacking logic; so @Charles-Gagnon @udeeshagautam if you add the respective webpack.config's to your extensions, it will webpack them before creating the vsix.